src/classes/BDI_DataImportService.cls

```diff
@@ -704,20 +704,54 @@ global with sharing class BDI_DataImportService {
     * from the list of DataImport Fields, which have the form DIObject.developerfieldname in the field's help text property.
     * @param dataImportObjectName the bdi object we care about (Contact1, Contact2, Account1, etc)
     * @param targetObjectName the underlying object the bdi object is (ie, Contact, Account, etc)
-    * @param listStrDIFields the DataImport fields to consider
+    * @param dataImportFields the DataImport fields to consider
     * @return Map<String, String> a map of DataImport fields to underlying object fields (eg, Donation_Amount__c => Amount)
     */
     global static Map<String, String> mapFieldsForDIObject(String dataImportObjectName, String targetObjectName, List<String> dataImportFields) {
 
+        BDISettings__c bdiSettings = BDISettings__c.getOrgDefaults();
+
         Map<String, String> dataImportFieldToTargetField = new Map<String, String>();
+        Map<String, String> bdiFIeldMappingMap = new Map<String, String>();
+
+        Boolean useCustomMetadatainsteadOfHelpText = false;
+
+        if (bdiSettings.UseCustomMetadatainsteadOfHelpText__c) {
+            system.debug('USING THE NEW STAFF');
+            BDI_Field_Mapping__mdt[] bdiFIeldMappingList = [SELECT
+                                                                Source_Field__c,
+                                                                Target_Field__c
+                                                            FROM
+                                                                BDI_Field_Mapping__mdt];
+
+            for (BDI_Field_Mapping__mdt customMetadata : bdiFIeldMappingList) {
+
+                bdiFIeldMappingMap.put(customMetadata.Source_Field__c.toLowerCase(), customMetadata.Target_Field__c);
+            }
+
+            useCustomMetadatainsteadOfHelpText = true;
+        }
 
         for (String dataImportField : dataImportFields) {
 
-            DescribeFieldResult fieldDescribe =
-                UTIL_Describe.getFieldDescribe(UTIL_Namespace.StrTokenNSPrefix('DataImport__c'), dataImportField);
-            String helpText = fieldDescribe.getInlineHelpText();
+            DescribeFieldResult fieldDescribe;
+            String helpText = '';
```

Comment:
helpText -> targetFieldName

I believe this variable is used to track the target field name and calling it help text is misleading, particularly when we're not using help text.

Reply:
I maintained the old name in order to keep the old logic as original as posible, but considering that this variable is not widely used in the old logic and that the name can be misleading I agree with you.

Reply:
Verified, and this change makes it much easier to read.